dcpy/builds/plan.py

```diff
@@ -179,11 +186,63 @@ def recipe_from_yaml(path: Path) -> Recipe:
         return recipe
 
 
-def plan(recipe_file: Path) -> Path:
-    logger.info("Planning recipe")
-    lock_file = recipe_file.parent / f"{recipe_file.stem}.lock.yml"
+def repeat_recipe_from_source_data_versions(
+    version: str,
+    source_data_versions: pd.DataFrame,
+    template_recipe_path: Path | None = None,
+) -> Recipe:
+    template_recipe_path_ = template_recipe_path or Path("recipe.yml")
```

Comment:
Any reason to have `template_recipe_path` be nullable? Thinking Path("recipe.yml") might cause problems depending on where this is invoked.

Reply:
Yeah, main reason was that this is that currently, the calls into the cli work down to here. Meaning the `recipe_file` arg makes its way down, so in the case that repeat is specified and no "template" provided, revert to the default.

I think though to your point, two things should happen
1 . use the DEFAULT_RECIPE constant that also feeds into other things
2. have a specific helpful message in case that this isn't found

By our conventions now there should generally be a "recipe.yml" where the code is running, but yeah don't want to turn that assumption into unclear failures

Reply:
Fixed in latest commits